tests/cpp/test_multidevice_pipeline.cpp

```diff
@@ -375,31 +374,33 @@ class PipelineTestStagedReduction
       public ::testing::WithParamInterface<SchedulingMode> {};
 
 // 1D staged reduction
-// Inputs: X[A,B,C]
+// Inputs: X[num_devices,B,C]
 TEST_P(PipelineTestStagedReduction, StagedReduction) {
   auto scheduling_mode = GetParam();
 
-  int num_devices = communicator_->size();
-  int A = num_devices;
-  int B = 8;
-  int C = 64;
-  std::vector<int64_t> unsharded_input_sizes = {A, B, C};
-  std::vector<int64_t> input_sizes(unsharded_input_sizes);
-  input_sizes[0] = 1;
+  const int num_devices = communicator_->size();
+  constexpr int B = 8;
+  constexpr int C = 64;
 
   FusionGuard fg(fusion.get());
-  TensorView* tv0 = makeConcreteTensor(unsharded_input_sizes);
+  // The first dimension is made symbolic so `tv_out->definition()` won't
+  // become a squeeze when num_devices == 1. This wouldn't be a problem for
+  // automatic mode. However, for the manual mode, the scheduling code below
+  // assumes `tv_out->definition()` can be lowered to communication. A squeeze
+  // can't.
+  TensorView* tv0 = TensorViewBuilder()
+                        .dtype(DataType::Float)
+                        .contiguity(true)
+                        .shape({-1, B, C})
+                        .build();
```

Comment:
@naoyam I believe this is the real fix. 

I could fix the problem in other ways, e.g., lowering SqueezeOp to communication or disallowing `sum` to generate `SqueezeOp` in the first place. The former doesn't seem to be worth the time because this error only happens in manual intra-device scheduling mode, which will eventually go away. The latter seems to have a large blast radius.